models/compilers/gcc_7_3_0_model.py

```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+from models.compilers.compiler_model import CompilerModel
+
+class CompilerModelImplementation(CompilerModel):
```

Comment:
we just need one module per compiler (gcc, clang) and let the version be dynamic via "compiler --version".